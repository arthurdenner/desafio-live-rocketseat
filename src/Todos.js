import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { isEmpty } from "lodash/fp";

const Todos = ({ todos }) =>
  isEmpty(todos) ? (
    <div className="emptyTodos">There is no itens in the list</div>
  ) : (
    <ul>{todos.map(todo => <li key={todo.id}>{todo.text}</li>)}</ul>
  );

Todos.propTypes = {
  todos: PropTypes.array.isRequired
};

const mapStateToProps = state => ({
  todos: state.todos
});

export default connect(mapStateToProps)(Todos);
