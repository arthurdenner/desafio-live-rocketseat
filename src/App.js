import React from "react";
import { Provider } from "react-redux";
import store from "./store";
import Todos from "./Todos";

const App = () => (
  <Provider store={store}>
    <div style={{ padding: "1em" }}>
      <h1>Desafio Live Rocket Seat</h1>
      <Todos />
    </div>
  </Provider>
);

export default App;
