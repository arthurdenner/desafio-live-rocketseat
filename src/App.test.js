import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import App from "./App";
import Todos from "./Todos";

describe("app component", () => {
  const wrapper = shallow(<App />);

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });

  it("should contain a h1 element", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });

  it("should contain a h1 element with the `Desafio Live Rocket Seat` title", () => {
    const h1 = wrapper.find("h1").first();

    expect(h1.text()).toEqual("Desafio Live Rocket Seat");
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain a todos component", () => {
    expect(wrapper.find(Todos).length).toEqual(1);
  });
});
