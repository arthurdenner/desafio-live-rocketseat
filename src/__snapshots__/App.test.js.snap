// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <div
        style={
                Object {
                        "padding": "1em",
                      }
        }
>
        <h1>
                Desafio Live Rocket Seat
        </h1>
        <Connect(Todos) />
</div>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Desafio Live Rocket Seat
</h1>,
          <Connect(Todos) />,
        ],
        "style": Object {
          "padding": "1em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Desafio Live Rocket Seat",
          },
          "ref": null,
          "rendered": "Desafio Live Rocket Seat",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "padding": "1em",
                            }
          }
>
          <h1>
                    Desafio Live Rocket Seat
          </h1>
          <Connect(Todos) />
</div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Desafio Live Rocket Seat
</h1>,
            <Connect(Todos) />,
          ],
          "style": Object {
            "padding": "1em",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Desafio Live Rocket Seat",
            },
            "ref": null,
            "rendered": "Desafio Live Rocket Seat",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
